File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    var answer = [];
    
    var sum= brown + yellow;
 
    for(let i = 3; i<= brown; i++){
        if( sum % i === 0){
            let weight = sum / i;
            
            if( (i-2) * (weight-2) === yellow){
               
                return [ weight, i];
            }
        }  
    }
    
    return answer;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int [2];
        
        int sum = brown + yellow;
        
        for(int i = 3 ; i <= sum ;i++){
            int col = i; //세로
            int row= sum / col; //가로 
            
            
            if(row <3){
                continue;
            }
            
            if(row >= col){
                 if((row-2)*(col-2) == yellow){
                     answer[0] = row;
                     answer[1] = col;
                         break;
                 }
            }
           
        }
        return answer;
    }
}